src/interface/sparse_pack_base.cpp

```diff
@@ -145,21 +162,11 @@ SparsePackBase SparsePackBase::Build(T *pmd, const PackDescriptor &desc) {
                       "Seems like this variable might not actually be allocated.");
                   if (desc.with_fluxes && pv->IsSet(Metadata::WithFluxes)) {
                     pack_h(1, b, idx) = pv->flux[1].Get(t, u, v);
-                    PARTHENON_REQUIRE(pack_h(1, b, idx).size() ==
-                                          pack_h(0, b, idx).size(),
-                                      "Different size fluxes.");
-                    if (ndim > 1) {
-                      pack_h(2, b, idx) = pv->flux[2].Get(t, u, v);
-                      PARTHENON_REQUIRE(pack_h(2, b, idx).size() ==
-                                            pack_h(0, b, idx).size(),
-                                        "Different size fluxes.");
-                    }
-                    if (ndim > 2) {
-                      pack_h(3, b, idx) = pv->flux[3].Get(t, u, v);
-                      PARTHENON_REQUIRE(pack_h(3, b, idx).size() ==
-                                            pack_h(0, b, idx).size(),
-                                        "Different size fluxes.");
-                    }
+                    pack_h(2, b, idx) = pv->flux[2].Get(t, u, v);
+                    pack_h(3, b, idx) = pv->flux[3].Get(t, u, v);
+                  }
+                  if (desc.flat) {
+                    coords_h(idx) = pmbd->GetBlockPointer()->coords_device;
                   }
```

Comment:
```suggestion
```

Reply:
I think this is potentially confusing that for regular backs `GetCoords` takes the block number as an argument while for flat packs it takes the variable flat index.

Reply:
I want to be able to make this call, as I think there's a clear use-case: i.e., you need coords for your stencil op. However, I would be willing to give `GetCoords` two different names?